docs/RFCs/19042021_self_signed_certificates.md

```diff
@@ -0,0 +1,240 @@
+Feature Name: Add a method to generate certificates without using Kubernetes CA
+ 
+Status: Draft
+ 
+Start Date: 19-04-2021
+ 
+Authors: @prafull01, @abhisek-dwivedi, @madhurnawandar
+ 
+# Summary
+ 
+This RFC proposes a method of deploying the CockroachDB Helm chart in a secure mode by generating certificates without using Kubernetes CA.  
+The new method allows users to specify their own CA or use self-generated CA.  
+It handles the creation of Node and Client certificates using the CA.  
+It allows rotation of CA, Node, and Client certificates.  
+It allows specifying a duration for the generated certificates
+ 
+# Motivation
+ 
+Currently, CockroachDB supports 3 ways of cert management
+* Built-in CSR's in Kubernetes: Depends on the user to approve CSR for all the generated certificates manually. CSR is no longer support with Certificates.k8s.io/v1 API and will be deprecated in Kubernetes v1.22. Many Kubernetes distributions like VMware Tanzu, EKS, etc are not allowing Kubernetes CA to sign the CSR's using Kubernetes CA.
+* Cert-manager: This is the most efficient way of managing certificates, but not everyone uses cert-manager to manage the certificates. Also using cert-manager may be overkill for dev/test environments
+* Manual: The user has to generate the certificates and provide them in the form of secrets. This method puts the overhead of certificate management leading to multiple manual steps for the user
+ 
+CockroachDB user needs a default mechanism of cert management which should work on all the k8s distributions without the need for manual intervention. While cert-manager fits into the requirement, it makes it CockroachDB deployment dependent on external application. This new method of cert-management satisfies the user requirement without the need for 3rd party software like cert-manager.
```

Comment:
Remove it. Makes CockroachDB deployment dependent...

Reply:
done